logger/logger.go

```diff
@@ -88,6 +96,22 @@ func getBaseFields(baseFields BaseFields) map[string]interface{} {
 	return initFields
 }
 
+// Instantiated checks if the logger is instantiated
+func Instantiated() bool {
+	return log != nil
+}
+
+// AddRequestID adds a request_id field that will be logged on every subsequent log in the application
+func AddRequestID(requestID string) {
+	clearLogFields()
+	log = log.AddRequestID(requestID)
+}
+
+// returns the log variable to its original state, clearing any previously added extra fields
+func clearLogFields() {
```

Comment:
Entendi o que esse método faz, mas qual foi o problema que você encontrou e gerou essa necessidade?

Reply:
toda vez que o request_id era adicionado (em cada requisição), era criado um novo campo "request_id" no log, independente se já tinha um campo com o msm nome ou n.

o zap logger n verifica se ja tem o campo, ele simplesmente adiciona um novo, aí tava duplicando o campo.

Reply:
pois é, eu tentei fazer isso de primeira de varias formas, mas o zap logger não tem suporte pra isso :/ 
como hoje não temos nenhum caso de uso de adicionar algum campo personalizado, isso não daria problema.

além de que acho dificil que alguem faça alguma feature parecida com essa, visto que até hoje não tinha nada.

mass n é impossível!
caso a pessoa for fazer algo assim, terá que se atentar nesse ponto realmente :(

Reply:
Entendido e obrigado pelo alinhamento. Acho que é um ponto importante, mas se o zap logger dificulta a vida e não temos um cenário onde o problema pode acontecer, dá pra seguir assim, com esse ponto de atenção mesmo.